test/store_unittest.cc

```diff
@@ -0,0 +1,288 @@
+}
+
+TEST_F(MetadataAgentStoreTest, ManyResourceIDsStore) {
+  SetupStore();
+  MonitoredResource mr("some_resource", {});
+  const char* ids_arr[] = {"id1", "id2", "id3"};
+  std::vector<std::string> resourceIds(ids_arr, std::end(ids_arr));
+  store->UpdateResource(resourceIds, mr);
+  EXPECT_EQ("some_resource", store->LookupResource("id1").type());
+  EXPECT_EQ("some_resource", store->LookupResource("id2").type());
+  EXPECT_EQ("some_resource", store->LookupResource("id3").type());
+}
+
+TEST_F(MetadataAgentStoreTest, ManyIdsAndResourcesStore) {
+  SetupStore();
+  MonitoredResource mr1("some_resource1", {});
+  MonitoredResource mr2("some_resource2", {});
+  const char* ids_arr1[] = {"id1", "id2", "id3"};
+  const char* ids_arr2[] = {"id-a", "id-b", "id-c"};
+  std::vector<std::string> resourceIds1(ids_arr1, std::end(ids_arr1));
+  store->UpdateResource(resourceIds1, mr1);
+  std::vector<std::string> resourceIds2(ids_arr2, std::end(ids_arr2));
+  store->UpdateResource(resourceIds2, mr2);
+  EXPECT_EQ("some_resource1", store->LookupResource("id1").type());
+  EXPECT_EQ("some_resource1", store->LookupResource("id2").type());
+  EXPECT_EQ("some_resource1", store->LookupResource("id3").type());
+  EXPECT_EQ("some_resource2", store->LookupResource("id-a").type());
+  EXPECT_EQ("some_resource2", store->LookupResource("id-b").type());
+  EXPECT_EQ("some_resource2", store->LookupResource("id-c").type());
+}
+
+TEST_F(MetadataAgentStoreTest, MultipleResourceIdsUpdateStore) {
+  SetupStore();
+  MonitoredResource mr("some_resource", {});
+  const char* ids_arr[] = {"id1", "id2"};
+  std::vector<std::string> resourceIds(ids_arr, std::end(ids_arr));
+  store->UpdateResource(resourceIds, mr);
+  EXPECT_EQ("some_resource", store->LookupResource("id1").type());
+  EXPECT_EQ("some_resource", store->LookupResource("id2").type());
+  const char* ids_arr2[] = {"id-a", "id-b"};
+  std::vector<std::string> resourceIds2(ids_arr2, std::end(ids_arr2));
+  store->UpdateResource(resourceIds2, mr);
+  EXPECT_EQ("some_resource", store->LookupResource("id-a").type());
+  EXPECT_EQ("some_resource", store->LookupResource("id-b").type());
+}
+
+TEST_F(MetadataAgentStoreTest, EmptyMetadataStore) {
+  SetupStore();
+  std::map<MonitoredResource, MetadataStore::Metadata> metadata_map =
+      GetMetadataMap();
+  EXPECT_TRUE(metadata_map.empty());
+}
+
+TEST_F(MetadataAgentStoreTest, EmptyMetadataNonEmptyResourceStore) {
+  SetupStore();
+  MonitoredResource mr("some_resource", {});
+  std::vector<std::string> resourceId(1, "abc123");
+  store->UpdateResource(resourceId, mr);
+  std::map<MonitoredResource, MetadataStore::Metadata> metadata_map =
+      GetMetadataMap();
+  EXPECT_TRUE(metadata_map.empty());
+}
+
+TEST_F(MetadataAgentStoreTest, OneMetadataEntryStore) {
+  SetupStore();
+  MonitoredResource mr("some_resource", {});
+  MetadataStore::Metadata m(
+      "default-version",
+      false,
+      time::rfc3339::FromString("2018-03-03T01:23:45.678901234Z"),
```

Comment:
Why not use `std::chrono::system_clock::now()`?

Reply:
Done.